worlds/messenger/client_setup.py

```diff
@@ -0,0 +1,167 @@
+                # os.makedirs(target, exist_ok=True)
+                # with urllib.request.urlopen(mono_kick_url) as download:
+                #     with ZipFile(io.BytesIO(download.read()), "r") as zf:
+                #         for member in zf.infolist():
+                #             zf.extract(member, path=target)
+                # installer = subprocess.Popen([os.path.join(target, "precompiled"),
+                #                               os.path.join(folder, "MiniInstaller.exe")], shell=False)
+                # os.remove(target)
+            else:
+                installer = subprocess.Popen([mono_exe, os.path.join(folder, "MiniInstaller.exe")], shell=False)
+        else:
+            installer = subprocess.Popen(os.path.join(folder, "MiniInstaller.exe"), shell=False)
+
+        failure = installer.wait()
+        if failure:
+            messagebox("Failure", "Failed to install Courier", True)
+            os.chdir(working_directory)
+            raise RuntimeError("Failed to install Courier")
+        os.chdir(working_directory)
+    
+        if courier_installed():
+            messagebox("Success!", "Courier successfully installed!")
+            return
+        messagebox("Failure", "Failed to install Courier", True)
+        raise RuntimeError("Failed to install Courier")
+
+    def install_mod() -> None:
+        """Installs latest version of the mod"""
+        assets = request_data(MOD_URL)["assets"]
+        if len(assets) == 1:
+            release_url = assets[0]["browser_download_url"]
+        else:
+            for asset in assets:
+                if "TheMessengerRandomizerAP" in asset["name"]:
+                    release_url = asset["browser_download_url"]
+                    break
+            else:
+                messagebox("Failure", "Failed to find latest mod download", True)
+                raise RuntimeError("Failed to install Mod")
+
+        mod_folder = os.path.join(folder, "Mods")
+        os.makedirs(mod_folder, exist_ok=True)
+        with urllib.request.urlopen(release_url) as download:
+            with ZipFile(io.BytesIO(download.read()), "r") as zf:
+                for member in zf.infolist():
+                    zf.extract(member, path=mod_folder)
+
+        messagebox("Success!", "Latest mod successfully installed!")
+
+    def available_mod_update(latest_version: str) -> bool:
+        """Check if there's an available update"""
+        latest_version = latest_version.lstrip("v")
+        toml_path = os.path.join(folder, "Mods", "TheMessengerRandomizerAP", "courier.toml")
+        with open(toml_path, "r") as f:
+            installed_version = f.read().splitlines()[1].strip("version = \"")
+
+        logging.info(f"Installed version: {installed_version}. Latest version: {latest_version}")
+        # one of the alpha builds
+        return not latest_version.isnumeric() or tuplize_version(latest_version) > tuplize_version(installed_version)
```

Comment:
I'm getting a prompt for a new mod update even though I already have the latest one. A SemVer string is not numeric (at least in Python 3.11) so `available_mod_update` always returns True.

![image](https://github.com/ArchipelagoMW/Archipelago/assets/35039/275f20e3-93e9-49a3-95a5-4f93476256de)